### ISSUE TITLE
Allow including images with relative paths in rendered pages (e.g. markdown)

### ISSUE THREAD

**Issue by s-ol** [jonashaag/klaus#244]
On github it is common to include images in the README.md, like so:

https://git.s-ol.nu/janko-keystep-rgb/blob/master/README.md

    ![](doc/wiring.png)

this currently creates the right markup:

    <img alt="" src="doc/wiring.png">

however the browser then request the image at `https://git.s-ol.nu/janko-keystep-rgb/blob/master/doc/wiring.png`, where he gets served the HTML page showing the image, instead of the raw image with the correct mime type.

I see two solutions:
1. Detect this situation when rendering the markup and change the `src` to point to `.../raw/master/doc/wiring.png`. This seems to be what github does.
2. Leave the renderer alone and look at the incoming request, when chromium loads the image it sends an `Accept: accept: image/webp,image/apng,image/*,*/*;q=0.8` header that clarifies that it is looking for an image, and even explicitly states that anything else (like `text/html`) will only partially satisfy (`q=0.8`). This could be detected and the file server raw in that case.

**Comment by jonashaag** [comment]
I agree that we should fix this and I'm happy with both ways, but strong preference for 1.) because it is much simpler (though maybe not easier to implement)

**Comment by cpressey** [comment]
For whatever it's worth, I faced the same problem, and took this approach to solving it:

Github honours `?raw=true` in requests for files; compare

* https://github.com/catseye/Dungeons-of-Ekileugor/blob/master/images/dungeons-of-ekileugor.png
versus
* https://github.com/catseye/Dungeons-of-Ekileugor/blob/master/images/dungeons-of-ekileugor.png?raw=true

So what I did was

* made sure all my embedded image links included `?raw=true` in their querystring
* made my copy of klaus understand it too: https://github.com/jonashaag/klaus/compare/master...catseye:redirect-to-raw-with-qsarg

Happy to open a PR with that change, but I suspect most users would prefer a "smarter" solution that doesn't require they add querystrings to their image links.

**Comment by jonashaag** [comment]
> but I suspect most users would prefer a "smarter" solution that doesn't require they add querystrings to their image links.

Indeed 8-)
